staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types.go

```diff
@@ -66,6 +72,24 @@ type CustomResourceDefinitionVersion struct {
 	// Storage flags the version as storage version. There must be exactly one
 	// flagged as storage version.
 	Storage bool `json:"storage" protobuf:"varint,3,opt,name=storage"`
+	// Schema describes the schema for CustomResource used in validation, pruning, and defaulting.
+	// Top-level and per-version schemas are mutually exclusive.
+	// Per-version schemas may not all be set to identical values (top-level validation schema should be used instead)
+	// This field is alpha-level and is only honored by servers that enable the CustomResourceWebhookConversion feature.
+	// +optional
+	Schema *JSONSchemaProps `json:"schema,omitempty" protobuf:"bytes,4,opt,name=schema"`
```

Comment:
My understanding is that for the top-level field, we were planning on having [a list of validation methods](https://github.com/kubernetes/kubernetes/blob/76234a31b0491134b6c08f65b7eeda376dcaf61f/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types.go#L220-L221) for CRs, so we defined a struct and made `openAPIV3Schema` nested. 

Now it becomes clear that we are not adding new methods to the `CustomResourceValidation`, so we simplify the per-version schema to just `JSONSchemaProps`. 

also cc @mbohlool @lavalamp on the related conversation: https://github.com/kubernetes/kubernetes/issues/67840

Reply:
> Now it becomes clear that we are not adding new methods to the `CustomResourceValidation`, so we simplify the per-version schema to just `JSONSchemaProps`.
> 
> also cc @mbohlool @lavalamp on the related conversation: #67840

#67840 actually proposes supporting a schema format other than openapi v3, which is exactly the use case for nesting openapi v3 as one of potentially multiple schema types. I think we should make the per-version schema structs in v1beta1 match the top-level structs, so that we can round-trip to whatever we end up with in v1, and have a path to transition to a different schema if that's where #67840 ends up. @lavalamp @sttts do you agree?

Reply:
Yes. In the KEP we proposed dropping the nested struct and going with just openapi v3, then in #67840 we brought up the idea around replacing openapi v3. I'm happy to go with the nested struct if we think that will make it easier to support another schema format, before we merge this PR.

Reply:
I am fine with keeping top level struct format with nested openapi v3. While last proposal on #67840 is to keep a more restricted version of openapi v3 but keeping the options open make more sense.

Reply:
More related context: https://github.com/kubernetes/kubernetes/issues/64483

I'm going to add a commit with
```
Schema *CustomResourceValidation `json:"schema,omitempty" protobuf:"bytes,4,opt,name=schema"`
```
first, which may look like:
```
spec:
  versions:
    - name: v1
      schema:
        openAPIV3Schema: ...       // covers schema, defaults and static validation
        anotherSchemaLanguage: ... // to be added in future?
      prune: true
```